### PR TITLE
Use SPR BM for e2e

### PR DESCRIPTION
**Pull request by tkatila**

### CODE REVIEW

**Comment by tkatila** [comment]
Please comment on the structure. In order to run multiple tests per run, I had to move the ginkgo variables out of the workflow. If the variables are for the whole test session, then qat is the only one that passes and dsa and iaa are skipped. This structure is a bit annoying as it ties device plugin project and the bmaas scripts more together. We wouldn't, for example, be able make a patch release for older releases without changing the bmaas side.

The bmaas changes are visible in the bmaas project.

**Comment by mythi** [comment]
@hj-johannes-lee can you comment?

**Comment by mythi** [comment]
> If the variables are for the whole test session, then qat is the only one that passes and dsa and iaa are skipped.

It's not clear to me what the issue is. I get a pretty comprehensive list with:
```
go test -v ./test/e2e/... -args -ginkgo.dry-run   -ginkgo.focus "Device:(dsa|iaa|qat|sgx)"
```

**Comment by tkatila** [comment]
When I tried to run qat, dsa and iaa e2e tests back to back, only qat executed due to the qat4's ginkgo focus/skip: `FOCUS="Mode:dpdk.*Resource:(cy|dc)" SKIP=App:crypto-perf`. dsa and iaa were skipped.

It might be possible to create a "e2e-spr" target which would run all the tests with one call. I can look into it.

**Comment by mythi** [comment]
> When I tried to run qat, dsa and iaa e2e tests back to back, only qat executed due to the qat4's ginkgo focus/skip: `FOCUS="Mode:dpdk.*Resource:(cy|dc)" SKIP=App:crypto-perf`. dsa and iaa were skipped.
> 
> It might be possible to create a "e2e-spr" target which would run all the tests with one call. I can look into it.

It's quite likely that the regexp creation is just broken. Moreover, we have #1687 WIP that should help streamline this a bit. That's why I asked @hj-johannes-lee to comment on this

**Comment by hj-johannes-lee** [comment]
sorry for being so slow recently.
Let me push a commit that fixes #1687. That would make this more easy..!

**Comment by tkatila** [comment]
This is the current ginkgo focus: `-ginkgo.focus "Device:(iaa|dsa)|Device:qat.*Mode:dpdk.*Resource:(cy|dc).*" -ginkgo.focus "Device:gpu.*Resource:i915"`

I think I tried it without Mode and Resource. For Resource it failed due to trying to run `generic` tests. I can't recall why Mode had issues, but I'm sure it had.

**Comment by hj-johannes-lee** [comment]
If mode was not used, then yeah, as you said
[Moide:dpdk] [Resource: generic] will be run.
And it is possible that [Mode:kernel] [Resource:cy1_dc0] will be also run.

The current focus seems ok to me. Why wasn't the ci/cd run?

**Comment by tkatila** [comment]
> The current focus seems ok to me regardless of the commit that I made. Why wasn't the ci/cd run?

There was/is an issue with the SPR machine. But it should be possible to retry it today.

**Comment by tkatila** [comment]
Flow passed now. It now runs QAT, IAA, DSA and GPU. SGX is wip in the HW.

File: .github/workflows/lib-e2e.yaml [hunk 1]
```diff
@@ -11,41 +11,20 @@ jobs:
...
-            runner: simics-spr
-            images: intel-iaa-plugin intel-idxd-config-initcontainer accel-config-demo intel-deviceplugin-operator
           - name: e2e-iaa-gnr
             targetjob: e2e-iaa
             runner: simics-gnr
             images: intel-iaa-plugin intel-idxd-config-initcontainer accel-config-demo intel-deviceplugin-operator
-          - name: e2e-qat
-            targetjob: e2e-qat FOCUS=Resource:generic
-            runner: qat
-            images: intel-qat-plugin intel-qat-initcontainer crypto-perf
```
[the remark this answers is not in the document]
Comment (tkatila) [reply .github/workflows/lib-e2e.yaml:40]: Got it, I'll revert that part.

**Review by mythi — APPROVED** [review]

**Pull request merged**